### ISSUE TITLE
Detect obsolescent and optional commands

### ISSUE THREAD

**Issue by stuff-and-nonsense** [koalaman/shellcheck#1875]
Unix commands can be categorized as either: obsolescent, optional or mandatory:
_https://en.wikipedia.org/wiki/List_of_Unix_commands_

I think obsolescent commands should be flagged as errors:
_fort77, qalter, qdel, qhold, qmove, qmsg, qrerun, qrls, qselect, qsig, qstat_

And optional commands should be warnings:
_admin, asa, bg, cc, c99, cal, cflow, compress, ctags, cxref, delta, ex, fc, fg, fuser, get, ipcrm, ipcs, jobs, lex, link, make, more, nl, nm, prs, rmdel, sact, sccs, strip, talk, ulimit, uncompress, unget, unlink, uucp, uustat, uux, val, vi, what, yacc, zcat_

The motivation is for better portability by relying only on external commands that are more likely to be present.

Minimal working example:
````
#!/usr/bin/env bash
set -Eeuo pipefail

# cmp is a mandatory unix command
cmp foo bar

# cal is an optional unix command
cal

# fort77 is an obsolete command
fort77
